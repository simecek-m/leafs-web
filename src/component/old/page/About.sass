@import "colors.sass"

#about-component
  color: $text-primary
  text-align: center

  h2
    color: $text-secondary
    text-transform: uppercase

  .section
    margin: 50px 0
    font-weight: bold
    font-size: 20px

  a
    color: $text-secondary
    text-decoration: none
    transition: 0.3s ease all

  a:hover
    color: $text-primary

  hr
    width: 80%
    max-width: 600px
    border-color: $text-secondary
    margin-bottom: 50px

  .author-link-icon
    margin: 0 5px

  .lottie-link-icon
    margin-right: 5px

