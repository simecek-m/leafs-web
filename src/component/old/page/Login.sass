@import "colors.sass"

.login-methods
  background: $background-secondary
  width: 40%
  max-width: 500px
  margin: 80px auto
  padding: 25px
  border-radius: 55px

  .login-method
    color: $text-secondary
    cursor: pointer
    height: 50px

    span
      &.authority
        display: inline-block
        color: $text-secondary
        text-decoration: none
        font-size: 28px
        font-weight: bold
        line-height: 50px
        vertical-align: middle

    .icon
      display: inline-block
      height: 40px
      width: 40px
      margin: 0 25px
      float: left
      clear: fix
