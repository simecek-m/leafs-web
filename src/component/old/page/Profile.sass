@import "colors.sass"

#profile-info
  display: block
  margin: 40px auto
  width: 80%
  min-width: 500px

  div
    &.row
      margin: 40px 0
      font-size: 22px
      font-weight: bold

      .property
        display: inline-block
        color: $text-secondary
        width: 45%
        text-align: right
        margin-right: 40px

      .value
        display: inline-block
        color: $text-primary
        width: 45%

      &.single
        width: 45%
        margin: 70px auto
        text-align: center

      img
        &.profile-picture
          display: inline-block
          width: 120px
          height: auto
          border-radius: 50%
