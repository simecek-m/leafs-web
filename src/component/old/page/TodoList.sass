@import "colors.sass"

.todo-list-component
  color: $text-primary

  .divider
    width: 70%
    min-width: 400px
    max-width: 700px
    margin: 60px auto 20px auto

  .tasks
    margin-left: 45%
    text-align: left
    user-select: none

