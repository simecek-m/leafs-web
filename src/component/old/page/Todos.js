import React from "react";
import { connect } from "react-redux";
import Lottie from "react-lottie";
import { withTranslation } from "react-i18next";
import { withRouter } from "react-router-dom";
import api, { setAuthorizationHeader } from "api";
import { faPlus } from "@fortawesome/free-solid-svg-icons";
import { showDangerNotificationWithStatus } from "notification";
import writeAnimation from "assets/animation/write.json";
import Title from "component/old/common/Title";
import Button from "component/old/button/Button";
import TodoListWidget from "component/old/todo/TodoListWidget";
import Loading from "component/old/animation/Loading";
import Settings from "component/old/menu/Settings";
import SettingsItem from "component/old/menu/SettingsItem";
import { faPlusCircle } from "@fortawesome/free-solid-svg-icons";
import { redirect } from "helper/router";
import { withModal } from "modal/withModal";
import { INPUT } from "modal/types";

export class Todos extends React.Component {
  state = {
    loading: true,
    error: false,
    list: []
  };

  componentDidMount() {
    const { user } = this.props;
    api
      .get("/lists", setAuthorizationHeader(user))
      .then(response =>
        this.setState({
          list: response.data,
          loading: false
        })
      )
      .catch(error => {
        showDangerNotificationWithStatus(
          error.response.data.message,
          error.response.status
        );
        this.setState({
          error: true,
          loading: false
        });
      });
  }

  deleteTodoList(id, index) {
    const { user } = this.props;
    api
      .delete(`/lists/${id}`, setAuthorizationHeader(user))
      .then(() => {
        const list = this.state.list;
        list.splice(index, 1);
        this.setState({
          list
        });
      })
      .catch(error =>
        showDangerNotificationWithStatus(
          error.response.data.message,
          error.response.status
        )
      );
  }

  addTodoList = title => {
    const { user } = this.props;
    const todoList = { title };
    api
      .post(`/lists`, todoList, setAuthorizationHeader(user))
      .then(response => {
        const list = this.state.list;
        list.push(response.data);
        this.setState({
          list
        });
      })
      .catch(error =>
        showDangerNotificationWithStatus(
          error.response.data.message,
          error.response.status
        )
      );
  };

  openNewTodoListModal() {
    const { t, openModal } = this.props;
    openModal({
      type: INPUT,
      title: t("todos.modal.title"),
      text: t("todos.modal.text"),
      onConfirm: this.addTodoList
    });
  }

  render() {
    const { t } = this.props;
    const items = this.state.list.map((item, index) => (
      <TodoListWidget
        onClick={() => redirect(`/lists/${item._id}`)}
        onDelete={() => this.deleteTodoList(item._id, index)}
        key={item._id}
        title={item.title}
        count={item.tasks.filter(task => !task.completed).length}
      />
    ));
    const writeAnimationOptions = {
      loop: true,
      autoplay: true,
      animationData: writeAnimation,
      rendererSettings: {
        preserveAspectRatio: "xMidYMid slice"
      }
    };
    const writeFirstTodoAnimation = (
      <div>
        <h2>{t("todos.createFirstList")}</h2>
        <Lottie
          options={writeAnimationOptions}
          width={400}
          height={300}
          style={{ marginBottom: "50px" }}
          isClickToPauseDisabled={true}
        />
        <Button
          icon={faPlus}
          text={t("todos.create")}
          onClick={() => this.openNewTodoListModal()}
        />
      </div>
    );
    const content = items.length > 0 ? items : writeFirstTodoAnimation;
    return (
      <div className="animated fadeIn">
        <Settings>
          <SettingsItem
            icon={faPlusCircle}
            onClick={() => this.openNewTodoListModal()}
          />
        </Settings>
        <Title text={t("todos.title")} />
        <Loading loading={this.state.loading} error={this.state.error}>
          {content}
        </Loading>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    user: state.user
  };
};

export default withModal(
  withRouter(connect(mapStateToProps)(withTranslation()(Todos)))
);
