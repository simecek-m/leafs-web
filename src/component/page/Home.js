import Topbar from "component/navigation/Topbar";

export function HomePage() {
  return (
    <div>
      <Topbar />
    </div>
  );
}
