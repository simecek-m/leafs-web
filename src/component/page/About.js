import React from "react";
import Title from "component/common/Title";
import { faInfoCircle } from "@fortawesome/free-solid-svg-icons";
import { withTranslation } from "react-i18next";
import Back from "component/navigation/Back";
import Settings from "component/menu/Settings";
import Lottie from "react-lottie";
import loadingAnimation from "assets/animation/loading.json";
import errorAnimation from "assets/animation/error.json";
import writeAnimation from "assets/animation/write.json";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faLink } from "@fortawesome/free-solid-svg-icons";
import { faGithub } from "@fortawesome/free-brands-svg-icons";
import "component/page/About.sass";

export function About({ t }) {
  const loadingAnimationOptions = {
    loop: true,
    autoplay: true,
    animationData: loadingAnimation,
    rendererSettings: {
      preserveAspectRatio: "xMidYMid slice"
    }
  };

  const errorAnimationOptions = {
    loop: true,
    autoplay: true,
    animationData: errorAnimation,
    rendererSettings: {
      preserveAspectRatio: "xMidYMid slice"
    }
  };

  const writeAnimationOptions = {
    loop: true,
    autoplay: true,
    animationData: writeAnimation,
    rendererSettings: {
      preserveAspectRatio: "xMidYMid slice"
    }
  };

  return (
    <div id="about-component" className="animated fadeIn">
      <Settings />
      <Back />
      <Title text={t("about.title")} icon={faInfoCircle} />
      <div className="section">
        <h2>{t("about.author")}</h2>
        <div>Martin Šimeček</div>
        <div className="links">
          <a
            href="https://github.com/simecek-m"
            rel="noopener noreferrer"
            target="_blank"
          >
            <FontAwesomeIcon className="author-link-icon" icon={faGithub} />
          </a>
          <a
            href="https://simecek.dev"
            rel="noopener noreferrer"
            target="_blank"
          >
            <FontAwesomeIcon className="author-link-icon" icon={faLink} />
          </a>
        </div>
      </div>
      <div className="section">
        <h2>{t("about.animations")}</h2>
        <div>
          {t("about.animationDescription", {
            name: "Skeleton UI",
            author: "Surani's Fun World"
          })}
        </div>
        <a
          href="https://lottiefiles.com/4463-hi"
          rel="noopener noreferrer"
          target="_blank"
        >
          <FontAwesomeIcon className="lottie-link-icon" icon={faLink} />
          Lottie Files
        </a>
        <Lottie
          options={loadingAnimationOptions}
          isClickToPauseDisabled={true}
          width={400}
          height={250}
        />
        <hr />
        <div>
          {t("about.animationDescription", {
            name: "Not Found",
            author: "Oscar Daniel Martínez Núñez"
          })}
        </div>
        <a
          href="https://lottiefiles.com/4339-not-found"
          rel="noopener noreferrer"
          target="_blank"
        >
          <FontAwesomeIcon className="lottie-link-icon" icon={faLink} />
          Lottie Files
        </a>
        <Lottie
          options={errorAnimationOptions}
          isClickToPauseDisabled={true}
          width={400}
          height={250}
        />
        <hr />
        <div>
          {t("about.animationDescription", {
            name: "Write",
            author: "kinoko"
          })}
        </div>
        <a
          href="https://lottiefiles.com/5729-write"
          rel="noopener noreferrer"
          target="_blank"
        >
          <FontAwesomeIcon className="lottie-link-icon" icon={faLink} />
          Lottie Files
        </a>
        <Lottie
          options={writeAnimationOptions}
          isClickToPauseDisabled={true}
          width={400}
          height={250}
        />
      </div>
    </div>
  );
}

export default withTranslation()(About);
