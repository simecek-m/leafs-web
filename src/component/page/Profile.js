import React from "react";
import { connect } from "react-redux";
import jsonwebtoken from "jsonwebtoken";
import Back from "component/navigation/Back";
import Button from "component/button/Button";
import { faDoorOpen } from "@fortawesome/free-solid-svg-icons";
import { logout } from "store/actions";
import Title from "component/common/Title";
import { faUserCircle } from "@fortawesome/free-solid-svg-icons";
import { withTranslation } from "react-i18next";
import Settings from "component/menu/Settings";
import "component/page/Profile.sass";

export function Profile({ user, logout, t }) {
  const userInfo = jsonwebtoken.decode(user);
  return (
    <div className="animated fadeIn">
      <Settings />
      <Back />
      <Title text={t("profile.title")} icon={faUserCircle} />
      <div id="profile-info">
        <div className="row">
          <span className="property">{t("profile.avatar")}</span>
          <img
            className="profile-picture"
            src={userInfo.picture}
            alt="profile"
          />
        </div>
        <div className="row">
          <span className="property">{t("profile.firstName")}</span>
          <span className="value">{userInfo.given_name}</span>
        </div>
        <div className="row">
          <span className="property">{t("profile.lastName")}</span>
          <span className="value">{userInfo.family_name}</span>
        </div>
        <div className="row">
          <span className="property">{t("profile.email")}</span>
          <span className="value">{userInfo.email}</span>
        </div>
        <div className="row">
          <span className="property">{t("profile.authority")}</span>
          <span className="value">{userInfo.iss}</span>
        </div>
        <div className="row single">
          <Button
            icon={faDoorOpen}
            text={t("profile.logout")}
            onClick={() => logout()}
          />
        </div>
      </div>
    </div>
  );
}

const mapStateToProps = state => {
  return {
    user: state.user
  };
};

export default connect(
  mapStateToProps,
  { logout }
)(withTranslation()(Profile));
