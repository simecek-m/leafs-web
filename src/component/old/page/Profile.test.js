import React from "react";
import { Profile } from "component/old/page/Profile";
import jsonwebtoken from "jsonwebtoken";
import { shallow } from "enzyme";

jest.mock("react-i18next", () => ({
  withTranslation: () => Component => {
    Component.defaultProps = { ...Component.defaultProps, t: key => key };
    return Component;
  }
}));

const TEST_USER = {
  picture: "url-avatar",
  given_name: "Tony",
  family_name: "Stark",
  email: "tony@stark.com",
  authority: "accounts.google.com"
};

const JWT_USER = jsonwebtoken.sign(TEST_USER, "secret");

describe("Profile component", () => {
  test("should render Profile page component", () => {
    const wrapper = shallow(<Profile user={JWT_USER} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should logout user after click LOGOUT button", () => {
    const logoutMock = jest.fn();
    const wrapper = shallow(<Profile user={JWT_USER} logout={logoutMock} />);
    const button = wrapper.find("#logout-button");
    expect(button).not.toBeNull();
    expect(logoutMock).not.toHaveBeenCalled();
    button.simulate("click");
    expect(logoutMock).toHaveBeenCalledTimes(1);
    expect(logoutMock).toHaveBeenCalledWith();
  });
});
