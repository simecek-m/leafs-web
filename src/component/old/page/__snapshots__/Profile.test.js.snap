// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component should render Profile page component 1`] = `
<div
  className="animated fadeIn"
>
  <Wrapper />
  <withRouter(Back) />
  <Title
    icon={
      Object {
        "icon": Array [
          496,
          512,
          Array [],
          "f2bd",
          "M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z",
        ],
        "iconName": "user-circle",
        "prefix": "fas",
      }
    }
    text="profile.title"
  />
  <div
    id="profile-info"
  >
    <div
      className="row"
    >
      <span
        className="property"
      >
        profile.avatar
      </span>
      <img
        alt="profile"
        className="profile-picture"
        src="url-avatar"
      />
    </div>
    <div
      className="row"
    >
      <span
        className="property"
      >
        profile.firstName
      </span>
      <span
        className="value"
      >
        Tony
      </span>
    </div>
    <div
      className="row"
    >
      <span
        className="property"
      >
        profile.lastName
      </span>
      <span
        className="value"
      >
        Stark
      </span>
    </div>
    <div
      className="row"
    >
      <span
        className="property"
      >
        profile.email
      </span>
      <span
        className="value"
      >
        tony@stark.com
      </span>
    </div>
    <div
      className="row"
    >
      <span
        className="property"
      >
        profile.authority
      </span>
      <span
        className="value"
      />
    </div>
    <div
      className="row single"
    >
      <Button
        icon={
          Object {
            "icon": Array [
              640,
              512,
              Array [],
              "f52b",
              "M624 448h-80V113.45C544 86.19 522.47 64 496 64H384v64h96v384h144c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16zM312.24 1.01l-192 49.74C105.99 54.44 96 67.7 96 82.92V448H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h336V33.18c0-21.58-19.56-37.41-39.76-32.17zM264 288c-13.25 0-24-14.33-24-32s10.75-32 24-32 24 14.33 24 32-10.75 32-24 32z",
            ],
            "iconName": "door-open",
            "prefix": "fas",
          }
        }
        id="logout-button"
        onClick={[Function]}
        text="profile.logout"
      />
    </div>
  </div>
</div>
`;
